ts/src/indodax.ts

```diff
@@ -852,23 +853,42 @@ export default class indodax extends Exchange {
          * @param {object} [params] extra parameters specific to the exchange API endpoint
          * @returns {object} an [order structure]{@link https://docs.ccxt.com/#/?id=order-structure}
          */
-        if (type !== 'limit') {
-            throw new ExchangeError (this.id + ' createOrder() allows limit orders only');
-        }
         await this.loadMarkets ();
         const market = this.market (symbol);
         const request: Dict = {
             'pair': market['id'],
             'type': side,
             'price': price,
         };
-        const currency = market['baseId'];
-        if (side === 'buy') {
-            request[market['quoteId']] = amount * price;
-        } else {
-            request[market['baseId']] = amount;
+        let priceIsRequired = false;
+        let quantityIsRequired = false;
+        if (type === 'market') {
+            let quoteAmount = undefined;
+            const precision = market['precision']['price'];
+            if (side === 'buy') {
+                if (price === undefined) {
+                    throw new InvalidOrder (this.id + ' createOrder() requires the price argument for market buy orders to calculate the total cost to spend (amount * price).');
+                }
+                const amountString = this.numberToString (amount);
+                const priceString = this.numberToString (price);
+                quoteAmount = Precise.stringMul (amountString, priceString);
+                request[market['quoteId']] = this.decimalToPrecision (quoteAmount, TRUNCATE, precision, this.precisionMode);
```

Comment:
instead of manually doing this you can call `costToPrecision`.

Additionally we should also support `cost` in `params` in this case (as an alternative to providing price+amount)

Reply:
Noted, I've updated based on your comments. please check again.

Reply:
@sungjinoh thanks!